### ISSUE TITLE
Update project name on new release if equivalent

### ISSUE THREAD

**Issue by di** [pypa/warehouse#4680]
Currently a project name is set when the project is created, and cannot be updated.

It should be possible to change the project name to something that will canonicalize to be the same name (e.g. from `CommonMark` to `commonmark`) by uploading a new release with the new name.

---

**Good First Issue**: This issue is good for first time contributors. If you've already contributed to Warehouse, work on [another issue without this label](https://github.com/pypa/warehouse/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+-label%3A%22good+first+issue%22) instead. If there is not a corresponding pull request for this issue, it is up for grabs. For directions for getting set up, see our [Getting Started Guide](https://warehouse.pypa.io/development/getting-started/). If you are working on this issue and have questions, feel free to ask them here, [`#pypa-dev` on Freenode](https://webchat.freenode.net/?channels=%23pypa-dev), or the [pypa-dev mailing list](https://groups.google.com/forum/#!forum/pypa-dev).

**Comment by kwasielew** [comment]
Hey @di , can I claim this ticket?

**Comment by di** [comment]
Hi @kwasielew, glad to hear you're working on this. We won't consider an issue "claimed" until there is a PR for it. I don't think anyone else is currently working on this though.

**Comment by di** [comment]
Blocked on #4772.

**Comment by pradyunsg** [comment]
@pganssle I removed this from the Project board, since it's blocked as @di noted above.

**Comment by di** [comment]
No longer blocked.
